model/src/main/java/com/basistech/rosette/apimodel/RelationshipsOptions.java

```diff
@@ -56,23 +63,53 @@ public void setAccuracyMode(AccuracyMode accuracyMode) {
         this.accuracyMode = accuracyMode;
     }
 
-    @Override
-    public int hashCode() {
-        return accuracyMode != null ? accuracyMode.hashCode() : 0;
-    }
-
     /**
-     * if the param is a {@code RelationshipOptions}, compare contents for equality
-     * @param o the object
-     * @return whether or not the param object is equal to this object
+     * @return the discoveryMode flag.
      */
+    public Boolean getDiscoveryMode() {
+        return discoveryMode;
+    }
+
     @Override
     public boolean equals(Object o) {
-        if (!(o instanceof RelationshipsOptions)) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
             return false;
         }
-
         RelationshipsOptions that = (RelationshipsOptions) o;
-        return accuracyMode != null ? accuracyMode.equals(that.getAccuracyMode()) : that.accuracyMode == null;
+        return discoveryMode == that.discoveryMode
+            && accuracyMode == that.accuracyMode;
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(accuracyMode, discoveryMode);
+    }
+
+    public static class Builder {
+        private Boolean discoveryMode;
+
+        public Builder() {
+            //
+        }
+
+
+        /**
+         * DocumentRequest discovery mode. If the value is {@code true}, then the the endpoint will open relationship
+         * extraction results. If {@code false}, not. If {@code null}, the endpoint will perform default processing.
+         * @param discoveryMode discovery mode
+         * @return this.
+         */
+        public RelationshipsOptions.Builder discoveryMode(Boolean discoveryMode) {
```

Comment:
I'm a bit confused about the usage of the builder. I can certainly expose that also.